lib/throttle.js

```diff
@@ -25,6 +25,21 @@ module.exports = (api) => {
     };
   };
 
+  api.metasync.debounce = (
+    timeout,
+    fn,
+    args
+  ) => {
+    let timer = null;
+    return () => {
+      clearTimeout(timer);
```

Comment:
I don't like the fact that `clearTimeout` is passed `null` during the first invocation. Can we set the first timer prior to returning the debounced function?